src/core/ext/transport/chaotic_good/frame.h

```diff
@@ -82,6 +82,9 @@ struct ClientFragmentFrame final : public FrameInterface {
     return stream_id == other.stream_id && EqHdl(headers, other.headers) &&
            end_of_stream == other.end_of_stream;
   }
+  // Shared ownership with transport.
+  std::shared_ptr<Arena> arena_;
+  void SetArena(std::shared_ptr<Arena> arena) { arena_ = arena; }
```

Comment:
Done.

Reply:
Actually I found that using `GetContext<Arena>()` is better here, so that we don't need to add an explicit constructor in the segment frames, and prevent distinguishing setting frames from segment frames during initialization.

Reply:
Thanks. Left a clarification question. I think you may need to pass the arena pointer to the activity.